financial-templates-lib/clients/ExpiringMultiPartyClient.js

```diff
@@ -91,31 +91,29 @@ class ExpiringMultiPartyClient {
   }
 
   async update() {
-    // Since this function can have a relatively long run-time, we should be safe and check contract state explicitly at the same block height.
-    // This way we won't experience errors where the contract state changes between calls.
-    this.latestBlock = await this.web3.eth.getBlockNumber();
-
-    this.collateralRequirement = this.toBN(
-      (await this.emp.methods.collateralRequirement().call(undefined, this.latestBlock)).toString()
-    );
-    this.liquidationLiveness = Number(await this.emp.methods.liquidationLiveness().call(undefined, this.latestBlock));
-
-    const events = await this.emp.getPastEvents("NewSponsor", { fromBlock: 0, toBlock: this.latestBlock });
+    // Fetch contract state variables in parallel.
+    const [collateralRequirement, liquidationLiveness, events, cumulativeFeeMultiplier] = await Promise.all([
+      this.emp.methods.collateralRequirement().call(),
+      this.emp.methods.liquidationLiveness().call(),
+      this.emp.getPastEvents("NewSponsor", { fromBlock: 0 }),
+      this.emp.methods.cumulativeFeeMultiplier().call()
+    ]);
+    this.collateralRequirement = this.toBN(collateralRequirement.toString());
+    this.liquidationLiveness = Number(liquidationLiveness);
     this.sponsorAddresses = [...new Set(events.map(e => e.returnValues.sponsor))];
+    this.cumulativeFeeMultiplier = this.toBN(cumulativeFeeMultiplier.toString());
 
-    // Fetch information about each sponsor.
-    const positions = await Promise.all(
-      this.sponsorAddresses.map(address => this.emp.methods.positions(address).call(undefined, this.latestBlock))
-    );
-    const collateral = await Promise.all(
-      this.sponsorAddresses.map(address => this.emp.methods.getCollateral(address).call(undefined, this.latestBlock))
-    );
+    // Fetch sponsor position, liquidation, and current time data in parallel.
+    const [positions, allLiquidations, currentTime] = await Promise.all([
+      Promise.all(this.sponsorAddresses.map(address => this.emp.methods.positions(address).call())),
```

Comment:
assuming bluebird promises here:
```
Promise.map(this.sponsorAddresses, address =>{
  return this.emp.methods.positions(address).call()
},{concurrency:20}))
```
And you can do the same thing for the line below. At least here you will have flow control doing a max of 40 calls in flight (20 per map)

Reply:
Thanks, I implemented this change! @mrice32 @chrismaree @daywiss PTAL